src/Sort.c

```diff
@@ -144,6 +156,182 @@ int quickSort(int arr[], int low, int high){
   return 0;
 }
 
+// QUICK SORT OF ARRAY OF 'ELEMENT'
+
+int partitionWithElementsByNumber(Element arr[], int low, int high){
+  // last Element becomes pivot
+  int pivot = (arr+high)->number;
+  // i is temporary pivot index - marks the index in the array, where everything with a smaller index (to the left)
+  // is already smaller than the pivot itself, but not sorted itself
+  int i = (low - 1);
+
+  // for all elements smaller than pivot move them to the left
+  for (int j = low; j < high; j++){
+    if ((arr+j)->number <= pivot){
+       // move temporary pivot index forward
+      i++;
+      // swap current element with the element at the temporary pivot
+      swapElements(arr+i, arr+j);
+    }
+  }
+  // move the pivot element to the correct pivot position (between smaller and larger elements)
+  swapElements(arr+i+1, arr+high);
+  return (i+1);
+}
+
+// will return 0 if first param is bigger than second
+// otherwise returns 1
+int isBiggerOrSameCharset(char charset[], char charset2[]) {
+  int result;
+  // checks all 3 letters in letters array for size
+  // TODO use global variable instead of '3'
+  for (int i = 0; i < 3; i++) {
```

Comment:
good solution i'm totally fine with, strcmp could  save a few lines, it also looks at strings character by character: https://www.programiz.com/c-programming/library-function/string.h/strcmp